### PR TITLE
feat(driver-adapters): automate npm publish

### PR DESCRIPTION
**Pull request by jkomyno**
This PR closes https://github.com/prisma/team-orm/issues/361.

### CODE REVIEW

File: .github/workflows/publish-driver-adapters.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        if: ${{ failure() }}
+        run: echo "SLACK_FOOTER=<$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID|Click here to go to the job logs>" >> $GITHUB_ENV
+
+      - name: Slack Notification on Failure
+        if: ${{ failure() }}
+        uses: rtCamp/action-slack-notify@v2.2.1
+        env:
+          SLACK_TITLE: "prisma-schema-wasm publishing failed :x:"
+          SLACK_COLOR: "#FF0000"
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_DRIVER_ADAPTERS_FAILING }}
```
Comment (jkomyno) [inline .github/workflows/publish-driver-adapters.yml:88]: TODO: add secret `SLACK_WEBHOOK_DRIVER_ADAPTERS_FAILING`. This should be similar to `SLACK_WEBHOOK_WASM_FAILING`.

**Comment by jkomyno** [comment]
Note: this PR can be tested locally via:

```sh
act -s GITHUB_TOKEN="$(gh auth token)" -j build -W .github/workflows/publish-driver-adapters.yml --input enginesWrapperVersion="alpha-0.0.1" --input engineHash="61d77d2871e6b839403a101940abdb0e12d8ee1d"
```

File: .github/workflows/publish-driver-adapters.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        if: ${{ failure() }}
+        run: echo "SLACK_FOOTER=<$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID|Click here to go to the job logs>" >> $GITHUB_ENV
+
+      # - name: Slack Notification on Failure
+      #   if: ${{ failure() }}
+      #   uses: rtCamp/action-slack-notify@v2.2.1
+      #   env:
+      #     SLACK_TITLE: "prisma-schema-wasm publishing failed :x:"
+      #     SLACK_COLOR: "#FF0000"
+      #     SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_DRIVER_ADAPTERS_FAILING }}
```
Comment (jkomyno) [inline .github/workflows/publish-driver-adapters.yml:91]: TODO: add back once the `SLACK_WEBHOOK_DRIVER_ADAPTERS_FAILING` secret is added to the repository.

File: .github/workflows/publish-driver-adapters.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        run: echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > ./.npmrc
+
+      - name: Create and publish versions
+        working_directory: query-engine/driver-adapters/js
+        uses: changesets/action@v1
+        with:
+          version: ${{ github.event.inputs.enginesWrapperVersion }})
+          commit: "chore(driver-adapters): update versions"
+          title: "chore(driver-adapters): update versions"
+          publish: pnpm publish --dry-run -r --access public --tag ${{ github.event.inputs.npmDistTag }}
```
Comment (jkomyno) [inline .github/workflows/publish-driver-adapters.yml:73]: TODO: remove `--dry-run` once we're sure the publish step works fine.

**Review by janpio — COMMENTED** [review]
Do I understand correctly that this would also publish the driver adapter packages with the same engines-wrapper like version a la `5.4.0-25.53d58ea0a4600a7f95eaeb1bff7b99c7d552671d`?

As these are packages installed by our users directly, vs something included in the CLI and Client or VS Code extension, I don't think that will necessarily work and be a good experience for the, We would also have no differentation between `dev` and real release version.

So if I understood this correctly, our assumptions here were wrong and should probably be revisited.

---

What the sprint document included and meant, was probably "only" to make sure that no one has to run publish to npm manually when we upgrade a driver adapter. The current `0.x` version is probably ok and even better for that until we figure out exactly how to deal with all this.

**Review by janpio — CHANGES_REQUESTED** [review]
On hold until https://github.com/prisma/team-orm/issues/386

**Pull request closed without merging**